schema_cache.rst

```diff
@@ -0,0 +1,119 @@
+.. _schema_cache:
+
+Schema Cache
+============
+
+There are several PostgREST features that need information from the database schema.
+If they accessed this information directly from the database every time they needed it, it would be too costly.
+That is why, when PostgREST starts, it generates a database schema cache and uses it to get the information needed for these features:
+
+- For :ref:`resource_embedding`, it looks for foreign key constraints to determine relationships between tables.
+- For :ref:`Database Functions <s_procs>`, it looks for their metadata: parameters, return type, volatility and `overloading <https://www.postgresql.org/docs/current/xfunc-overload.html>`_.
+- For :ref:`Upserts <upsert>`, it looks for the tables' primary keys.
+- For :ref:`insert_update`, it also looks for the tables' primary keys in order to return the Location header.
+- For OPTIONS requests, it verifies the existence of tables and views and looks for INSTEAD OF triggers for the latter.
+- For :ref:`open-api`, it looks for information about tables (columns, primary keys and foreign keys), views (columns and INSTEAD OF triggers) and functions (metadata).
```

Comment:
@laurenceisla Do you think this part would look better as a table? We might be able to reduce some repeated words by putting a column for "feature" and another column for "required metadata".

Reply:
I agree, it should make it look less cluttered.

Reply:
> ... and another column for "required metadata".

I used "required schema cache information" instead, I think that way it should include everything in the list, WDYT?

Reply:
@laurenceisla The table looks much better!

Regarding "required schema cache information", I think "required metadata" is more precise since the metadata is not always in the cache, the schema cache might have to be reloaded to get it.

Reply:
> Regarding "required schema cache information", I think "required metadata" is more precise since the metadata is not always in the cache, the schema cache might have to be reloaded to get it.

Ah, I get it now. I agree.